megamek/src/megamek/common/actions/WeaponAttackAction.java

```diff
@@ -366,7 +366,8 @@ private static ToHitData toHitCalc(Game game, int attackerId, Targetable target,
                         || atype.countsAsFlak()
         );
 
-        boolean isIndirect = (wtype.hasModes() && weapon.curMode().equals(Weapon.MODE_MISSILE_INDIRECT));
+        boolean isIndirect = (weapon.hasModes() && (weapon.curMode().equals(Weapon.MODE_MISSILE_INDIRECT)
```

Comment:
Do we want to encapsulate this into a convenience function? I'm seeing it in at least two places.

Reply:
I thought about doing that after I had written it, but decided to leave it as is instead of going back and finding those occurrences. Since you've commented on it as well I've gone back and made the change.